tests/testsuite/hints.rs

```diff
@@ -32,6 +32,21 @@ fn empty_hints_no_warn() {
 
 #[cargo_test]
 fn unknown_hints_warn() {
+    Package::new("bar", "1.0.0")
```

Comment:
nit: test changes like this should be made before any of the feature work so that we can tell just by looking at the tests what the behavior change is.  By changing the test, it obscures that.

(granted, in some cases, changing the test is part of showcasing the feature kind of like the `rerun_untracks_other_files` test in a24abd6fc2347b3c0de0a96be40bf98df4ea030f

Reply:
I don't think doing so provided much value in this case, but nonetheless, done.